src/js/components/twemoji.js

```diff
@@ -0,0 +1,246 @@
+    matchIndex = match.index;
+    if (matchIndex !== textIndex) {
+      result.push(text.slice(textIndex, matchIndex).replace(vs16RegExp, ''));
+    }
+
+    rawText = match[0];
+    iconId = grabHEXCodePoint(rawText);
+    textIndex = matchIndex + rawText.length;
+
+    const randomNumber = Math.random();
+
+    /**
+     * A unique key is added to ensure the image component
+     * is removed if user the removes the emoji
+     * in the actual text.
+     * format: `twemoji-1f351-12-0.6236104024`.
+     */
+    const key = `twemoji-${iconId}-${textIndex}-${randomNumber}`;
+
+    result.push(<TwemojiImage key={key} style={style} rawText={rawText} iconId={iconId} />);
+    modified = true;
+  }
+
+  if (modified) {
+    // any text left to be added
+    if (textIndex < text.length) {
+      result.push(text.slice(textIndex).replace(vs16RegExp, ''));
+    }
+  } else {
+    result.push(text);
+  }
+
+  return result;
+};
+
+const flatten = (array) => {
+  let parsedChildren = [],
+    hasObjectType = false;
+
+  array.forEach((item) => {
+    if (Array.isArray(item)) {
+      parsedChildren = parsedChildren.concat(item);
+    } else {
+      parsedChildren.push(item);
+    }
+  });
+
+  hasObjectType = parsedChildren.some((x) => isObject(x));
+
+  return { parsedChildren, hasObjectType };
+};
+
+const parseChildren = (source, style) => {
+  if (!Array.isArray(source)) source = [source];
+  return flatten(
+    source.map((x) => {
+      return isString(x) ? parseString(x, style) : x;
+    }),
+  );
+};
+
+// Components for export
+
+/**
+ * Returns the equivalent Twemoji for a given emoji.
+ * If there is no equivalent, it will return the same emoji
+ * wrapped in text view
+ */
+export const Twemoji = ({ children, imageProps, ...props }) => {
+  const textStyle = StyleSheet.flatten(props.style);
+  const { parsedChildren } = parseChildren(children, { imageProps, textStyle });
+
+  if (parsedChildren.length && isObject(parsedChildren[0])) {
+    return parsedChildren[0];
+  }
+
+  return <Text {...props}>{children}</Text>;
+};
+
+/**
+ * Parse the children and replace the emoji with Twemoji.
+ * If there is no equivalent, it will return the same emoji.
+ */
+export const TwemojiText = ({ children, imageProps, ...props }) => {
+  const textStyle = StyleSheet.flatten(props.style);
+  const { parsedChildren, hasObjectType } = parseChildren(children, { imageProps, textStyle });
+
+  // If there in no Twemoji in the text, render it as it is.
+  if (!hasObjectType) {
+    return <Text {...props}>{parsedChildren.join('')}</Text>;
+  }
+
+  return (
+    <Text {...props}>
+      {/* Hack to retain the line height and to render emoji (images) within the line height*/}
+      <Text
+        key={'twemoji-line-height-retainer'}
+        accessible={false}
+        selectable={false}
+        style={{ color: 'transparent', fontSize: 0.1 }}
+      >
+        {' '}
+      </Text>
+      {parsedChildren}
+    </Text>
+  );
+};
```

Comment:
#### Reason to add hack/workaround:
There should be a string at the beginning to make sure the image (emoji) is aligned with the baseline of the text.

<img width="330" src="https://github.com/status-im/status-mobile/assets/19339952/e4291db2-a7f6-4b45-b4a7-921ac3432a3a">